### PR TITLE
50 post details page

### LINKED ISSUE #50: Post details page
Users should be able to view a post by clicking on it, which takes them to a unique page showing only that post, as well as its replies (and perhaps reposts) below it.

### PR DESCRIPTION
**Pull request by melleg**
Closes #50

### CODE REVIEW

File: backend/src/main/java/nl/itvitae/twitbook/post/PostService.java [hunk 1]
```diff
@@ -31,6 +33,8 @@ public Optional<Post> findByTypeAndLinkedPostAndPoster_UsernameIgnoreCase(Post.P
     return postRepository.findByTypeAndLinkedPostAndPoster_UsernameIgnoreCase(postType, linkedPost, username);
   }
 
+
+
```
Comment (mio-xenoblade) [inline backend/src/main/java/nl/itvitae/twitbook/post/PostService.java:41]: random whitespace

File: frontend/src/post/PostCard.tsx [hunk 2]
```diff
@@ -221,14 +265,23 @@ const PostCard: React.FC<PostCardProps> = ({ post: postProp }) => {
...
   if (deleted)
     return (
       <p className="py-2 px-4 glass rounded-lg text-light">Post deleted</p>
     );
 
   return (
-    <div className="py-2 pl-20 pr-4 glass relative rounded-lg">
+    <div
+      onClick={(e) => tryNavigate(e, getPost()!.id)}
+      className={`py-2 pl-20 pr-4 glass relative rounded-lg ${className ?? ""}`}
```
Comment (mio-xenoblade) [inline frontend/src/post/PostCard.tsx]: cursor-pointer

**Review by NolKeereweer — COMMENTED** [review]
Replying to a post on the post details page doesn't seem to refresh it until page-refresh.
Other than that it looks very nice.

File: backend/src/main/java/nl/itvitae/twitbook/post/PostRepository.java [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
 public interface PostRepository extends JpaRepository<Post, Long> {
     List<Post> findByPoster_UsernameIgnoreCase(String username);
-
+    Page<Post> findByLinkedPost_Id(Long id, Pageable pageable);
```
Comment (mio-xenoblade) [inline backend/src/main/java/nl/itvitae/twitbook/post/PostRepository.java]: also gets reposts, when i assume this is to get replies only

**Review by mio-xenoblade — CHANGES_REQUESTED** [review]
Posts page also shows reposts as replies, but otherwise great code

File: backend/src/main/java/nl/itvitae/twitbook/post/PostController.java [hunk 4]
```diff
@@ -42,7 +35,7 @@ public class PostController {
   private final LikeRepository likeRepository;
   private final HashtagService hashtagService;
 
-  private static final int PAGE_SIZE = 4;
+  private static final int PAGE_SIZE = 20;
```
Comment (mio-xenoblade) [inline backend/src/main/java/nl/itvitae/twitbook/post/PostController.java:38]: frontend page only sometimes scrolls back to top

**Review by mio-xenoblade — APPROVED** [review]
scroll only sometimes resets

**Review by NolKeereweer — APPROVED** [review]
Everything looks alright. During testing found that post-details page doesn't refresh on reply but that can be a feature.

**Pull request merged**